sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredential.java

```diff
@@ -56,13 +58,15 @@ public class ChainedTokenCredential implements TokenCredential {
     private static final ClientLogger LOGGER = new ClientLogger(ChainedTokenCredential.class);
     private final List<TokenCredential> credentials;
     private final String unavailableError = this.getClass().getSimpleName() + " authentication failed. ---> ";
+    private AtomicReference<TokenCredential> cachedWorkingCredential;
```

Comment:
```suggestion
    private AtomicReference<TokenCredential> selectedCredential;
```

maybe this instead?

Reply:
"working" implies "non-working" somewhere which while true isn't verbiage we want maybe. :)

Reply:
Good point.